tests/objectstore/test_classic_metastore.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+
+from openghg.objectstore import get_object_from_json
+from openghg.objectstore.metastore import open_metastore
+from openghg.types import MetastoreError
+
+
+def test_metastore_read_write_mode(tmp_path):
```

Comment:
I can put the path in with the others, but the particular path isn't really important for these tests.

We might want to use the pytest fixtures rather than tempfile, since the pytest ones persist after the tests (until the next test run?) whereas the tempfile deletes them immediately.

Reply:
I just think for the tests we don't want to be creating temporary files/locations all over the user's system - I don't mind where this is done. If creating this using in build `pytest` features like `tmp_path` then that's fine but we don't want to approach this in 2 different ways really.

Reply:
I haven't addressed this in the latest PR. For unit tests on isolated components, I think `tmp_path` makes sense.

For more involved tests, mocking the actual OpenGHG set up makes sense. Keeping the file path names in the top level conftest,py makes sense as well, although creating the temporary directories at the module level would mean that we can run the tests in parallel. (Any tests that don't actually modify the object store could use data added at the session level though.)

Reply:
> I just think for the tests we don't want to be creating temporary files/locations all over the user's system - I don't mind where this is done. If creating this using in built `pytest` features like `tmp_path` seems like the best approach then that's fine but we don't want to implement this in 2 different ways really.

Sorry I didn't see this before I commented. You can set the location where pytest puts temp paths and directories; I don't know if I can make it the same as where the test stores are kept.

If I switch over the existing conftest files to use `tmp_path_factory` (which makes directories that persist with session level scope), then we'd have the option to use `tmp_path` and still keep all of the temporary files in the same location.

The files created are labelled pytest-NUM where NUM is incremented with each run. These get cleaned up eventually, and only a few persist at a time. This sounds useful to me because sometimes to debug things I have to put in a breakpoint, then find the temporary path, then look at the temporary directory while the debugger is running.

It is possible to only keep at most one temporary directory at a time: https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#base-temporary-directory

To avoid using `tmp_path` for this PR, I could create a function scope fixture in the top level conftest.py that provides a directory in the `tempfile` temp dir, and then clears it after the test runs. That would do the same thing that `tmp_path` is doing.

I think refactoring the tests is outside the scope of this PR, but another option is to leave in `tmp_dir` and open an issue to refactor the tests to use `tmp_dir` and `tmp_dir_factory` instead of pathlib and tempfile. (I'm not super keen on doing this, but it would be useful for unit tests where we want to test things in isolation. The other option for doing this is making an in-memory implementation of the interface for testing. In this case, `tmp_dir` was easier.)